packages/cryptoplease/lib/features/outgoing_split_key_payments/src/bl/oskp_verifier.dart

```diff
@@ -5,26 +5,40 @@ import 'package:rxdart/rxdart.dart';
 import 'package:solana/dto.dart';
 import 'package:solana/solana.dart';
 
+import '../../../../core/solana_helpers.dart';
 import 'outgoing_split_key_payment.dart';
 import 'repository.dart';
 
 @injectable
 class OSKPVerifier {
-  OSKPVerifier(this._client, this._repository);
+  OSKPVerifier(
+    this._client,
+    this._repository, {
+    @factoryParam required Ed25519HDPublicKey userPublicKey,
+  }) : _userPublicKey = userPublicKey;
 
   final SolanaClient _client;
   final OSKPRepository _repository;
+  final Ed25519HDPublicKey _userPublicKey;
 
   final Map<String, StreamSubscription<void>> _subscriptions = {};
   StreamSubscription<void>? _repoSubscription;
 
   void init() {
     _repoSubscription = _repository.watchWithReadyLinks().listen((payments) {
       for (final payment in payments) {
-        void onSuccess(String txId) {
-          final newStatus = OSKPStatus.success(txId: txId);
-          _repository.save(payment.copyWith(status: newStatus));
-          _subscriptions[payment.id]?.cancel();
+        Future<void> onSuccess(String txId) async {
+          final newStatus = await _client.getDestinations(txId).then(
```

Comment:
What if this method fails?

Retrieve the full tx in `_createStream` instead – it has error processing.